rust/operator-binary/src/security/authentication.rs

```diff
@@ -160,35 +189,74 @@ impl NifiAuthenticationConfig {
                     cb.add_volume_mount(STACKABLE_ADMIN_USERNAME, STACKABLE_USER_VOLUME_MOUNT_PATH);
                 }
             }
-            Self::Ldap(ldap) => {
-                ldap.add_volumes_and_mounts(pod_builder, container_builders)
+            Self::Ldap { provider } => {
+                provider
+                    .add_volumes_and_mounts(pod_builder, container_builders)
                     .context(AddLdapVolumesSnafu)?;
             }
+            Self::Oidc { provider, nifi, .. } => {
+                let admin_volume = Volume {
+                    name: STACKABLE_ADMIN_USERNAME.to_string(),
+                    secret: Some(SecretVolumeSource {
+                        secret_name: Some(build_oidc_admin_password_secret_name(nifi)),
+                        optional: Some(false),
+                        items: Some(vec![KeyToPath {
+                            key: STACKABLE_ADMIN_USERNAME.to_string(),
+                            path: STACKABLE_ADMIN_USERNAME.to_string(),
+                            ..KeyToPath::default()
+                        }]),
+                        ..SecretVolumeSource::default()
+                    }),
+                    ..Volume::default()
+                };
+                pod_builder.add_volume(admin_volume);
+
+                let container_builders = container_builders
+                    .into_iter()
+                    .map(|cb| {
+                        cb.add_volume_mount(
+                            STACKABLE_ADMIN_USERNAME,
+                            STACKABLE_USER_VOLUME_MOUNT_PATH,
+                        )
+                    })
+                    .collect();
```

Comment:
optional: In case you use this you need to mark `container_builders` as `mut`
```suggestion
                container_builders.iter_mut().for_each(|cb| {
                    cb.add_volume_mount(STACKABLE_ADMIN_USERNAME, STACKABLE_USER_VOLUME_MOUNT_PATH);
                });
```